packages/driver/src/cy/commands/actions/select.coffee

```diff
@@ -59,6 +59,22 @@ module.exports = (Commands, Cypress, cy, state, config) ->
       if not multiple and valueOrText.length > 1
         $utils.throwErrByPath "select.invalid_multiple"
 
+      ## deselect all options if we receive an empty array for a multi select or empty string for a single select
+      if valueOrText.length == 0 or (valueOrText.length == 1 and valueOrText[0] == "")
+        clearOptions = ->
+          options.$el.val([])
+
+        retryClear = ->
+          Promise
+            .try(clearOptions)
+            .catch (err) ->
+              options.error = err
+
+              cy.retry(retryClear, options)
+
+        return Promise
+          .try(retryClear)
```

Comment:
i'm not sure this Promise.try is necessary, can't this just be `return retryClear()`?

Reply:
Good catch! I initially had it in there in case I needed to fire any click or change events in a subsequent then, but forgot to take it out.